keg/generator.py

```diff
@@ -0,0 +1,82 @@
+# Copyright (c) 2020 SUSE Software Solutions Germany GmbH. All rights reserved.
+#
+# This file is part of keg.
+#
+# keg is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# keg is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with keg. If not, see <http://www.gnu.org/licenses/>
+#
+from jinja2 import Environment, FileSystemLoader
+import os
+
+from keg import (
+    image_definition,
+    kiwi_description,
+    template_functions
+)
+from keg.exceptions import KegError, KegKiwiValidationError
+
+
+def create_image_description(image_source,
+                             recipes_root,
+                             dest_dir,
+                             log,
+                             force=False):
+    """
+    Create a KIWI image description in dest_dir from image_source.
+
+    Expects a keg-recipes structure at recipes_root and image_source
+    referring to a directory under images in recipes_root.
+    """
+    # Sanity check image source location
+    if not os.path.exists(os.path.join(recipes_root, 'images', image_source)):
```

Comment:
I we make the assumption that the tree lives in git then this implies that we expect the user to have pulled the git repo and has it on their system. This is fair, question I have is whether we should build in a convenience that allows the code to pull the git repo?

- Pro for pulling the git repo is convenience for the user
- Con is that we increase our dependency structure
- Con we'd favor a specific VCS/SCM system (things other than git still exist)

Reply:
If you use keg manually you may forget to update your tree and work on outdated data, so I guess it wouldn't be just added convenience. Having this as at least an optional feature could be nice.

Reply:
Yes, I think RCS awareness will be a nice feature. We have to design some plugin system such that we do not force the use of a specific RCS system. We would write a git implementation as that is what we use.